util/osuser/user.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package osuser implements OS user lookup. It's a wrapper around os/user that
+// works on non-cgo builds.
+package osuser
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"tailscale.com/version/distro"
+)
+
+// LookupId is like os/user.LookupId but handles a few edge cases like gokrazy
+// and non-cgo lookups, and returns the user shell.
+func LookupId(uid string) (u *user.User, shell string, err error) {
```

Comment:
LookupId is a historical spelling mistake we can't fix now, but we wish it had really been LookupID per https://github.com/golang/go/wiki/CodeReviewComments#initialisms

There are arguments for either way here (match os/user vs use modern style). Your call.